src/data/complex/module.lean

```diff
@@ -32,9 +34,48 @@ noncomputable theory
 
 namespace complex
 
-instance algebra_over_reals : algebra ℝ ℂ := (complex.of_real).to_algebra
+variables {R : Type*} {S : Type*}
 
-@[simp] lemma smul_coe {x : ℝ} {z : ℂ} : x • z = x * z := rfl
+section
+
+variables [has_scalar R ℝ]
+
+instance : has_scalar R ℂ :=
+{ smul := λ r x, ⟨r • x.re, r • x.im⟩ }
+
+lemma smul_re' (r : R) (z : ℂ) : (r • z).re = r • z.re := rfl
+lemma smul_im' (r : R) (z : ℂ) : (r • z).im = r • z.im := rfl
+
+@[simp] lemma smul_coe {x : ℝ} {z : ℂ} : x • z = x * z :=
+by ext; simp [smul_re', smul_im']
+
+end
+
+instance [has_scalar R ℝ] [has_scalar S ℝ] [smul_comm_class R S ℝ] : smul_comm_class R S ℂ :=
+{ smul_comm := λ r s x, ext (smul_comm _ _ _) (smul_comm _ _ _) }
+
+instance [has_scalar R S] [has_scalar R ℝ] [has_scalar S ℝ] [is_scalar_tower R S ℝ] :
+  is_scalar_tower R S ℂ :=
+{ smul_assoc := λ r s x, ext (smul_assoc _ _ _) (smul_assoc _ _ _) }
+
+instance [monoid R] [mul_action R ℝ] : mul_action R ℂ :=
+{ one_smul := λ x, ext (one_smul _ _) (one_smul _ _),
+  mul_smul := λ r s x, ext (mul_smul _ _ _) (mul_smul _ _ _) }
+
+
```

Comment:
Is there a reason for this extra newline?

Reply:
No